File: app/src/main/java/com/thibaudperso/sonycamera/timelapse/model/IntervalometerSettings.java
package com.thibaudperso.sonycamera.timelapse.model;

import java.io.Serializable;


public class IntervalometerSettings implements Serializable {

    public int initialDelay = 0;
    public int intervalTime = 0;

    /**
     * 0 for infinity
     */
    public int framesCount = 0;


    public boolean isUnlimitedMode() {
        return framesCount == 0;
    }

    @Override
    public String toString() {
        return "IntervalometerSettings{" +
                "initialDelay=" + initialDelay +
                ", intervalTime=" + intervalTime +
                ", framesCount=" + framesCount +
                '}';
    }

}

File: app/src/main/java/com/thibaudperso/sonycamera/timelapse/model/IntervalometerSettings.kt
package com.thibaudperso.sonycamera.timelapse.model

import java.io.Serializable

data class IntervalometerSettings(var initialDelay: Long = 0L,
                                  var intervalTime: Long = 0L,
                                  /**
                                   * 0 for infinity
                                   */
                                  var framesCount: Int = 0) : Serializable {

    val isUnlimitedMode: Boolean
        get() = framesCount == 0

}
